### PR TITLE
dockerTools.*: Assertion against building for Darwin

### PR DESCRIPTION
**Pull request by infinisil**
###### Motivation for this change

Building a docker image with Darwin binaries just yields a confusing
error when ran:

    standard_init_linux.go:211: exec user process caused "exec format error"

This change prevents people from building such images in the first place

This PR is sponsored by Niteo :sparkles: 

Ping @grahamc @zupo 

###### Things done

- [x] Verified that the assertion is not triggered on Linux
- [x] Verified that the assertion is triggered on macOS

### CODE REVIEW

**Comment by matthewbauer** [comment]
Does it also need to be x86_64 architecture?

**Comment by Infinisil** [comment]
According to https://stackoverflow.com/a/52845392/6605742 it doesn't, so I now switched to `buildPlatform.isLinux`

**Review by zupo — APPROVED** [review]

**Comment by Infinisil** [comment]
Question to @grahamc: Is it possible to build a Linux docker image on macOS by doing it inside a Linux qemu VM? Because then this assertion might prevent people from doing that.

**Comment by matthewbauer** [comment]
If you change this from buildPlatform to hostPlatform, it should allow you to still cross compile a kernel from macOS to Linux and put it in a Docker image.

**Comment by Infinisil** [comment]
Just noticing the different platforms here too, including `darwin`: https://github.com/moby/moby/blob/master/image/spec/v1.2.md#image-json-field-descriptions

**Comment by tomberek** [comment]
It should remain possible to build the container on OSX, but get all binaries from cache or a nix-docker builder.

**Comment by Infinisil** [comment]
Ah so the above link is about `moby`, which is like a container abstraction thing, the `os` parameter doesn't matter for Docker since Docker can be installed on all operating systems. The `architecture` parameter does matter for Docker, as I believe it needs to be the same between the linux binaries within docker and the architecture docker runs on. But anyways, this doesn't seem relevant for this PR, docker builds linux containers, and that's what this PR restricts it to.

I initially confused hostPlatform and buildPlatform, I switched it around so that it's correct.

**Comment by grahamc** [comment]
Note that Windows Docker images can be built for Windows, executing Windows binaries, too.

**Comment by grahamc** [comment]
(_NO_ idea if anybody is using dockerTools to do that, so I wouldn't hold this PR back on that basis ... but maybe this PR would get reverted on that basis :) )

**Comment by Infinisil** [comment]
I'll go ahead and merge this soon if nobody has any complaints

**Pull request merged**
